# 2. IPYNB análisis dataset Fortnite

In [122]:
import pandas as pd

print(f'Pandas version: {pd.__version__}')

Pandas version: 3.0.0


### 2.1 — Exploración inicial

Muestra .head(), .tail(), .shape y .describe()

In [123]:
fortnite = pd.read_csv('datasets/Fortnite Statistics.csv')

display(fortnite.head())
display(fortnite.tail())
display(fortnite.shape)
display(fortnite.describe())

,Date,Time of Day,Placed,Mental State,Eliminations,Assists,Revives,Accuracy,Hits,Head Shots,Distance Traveled,Materials Gathered,Materials Used,Damage Taken,Damage to Players,Damage to Structures
0,4/10,6:00 PM,27,sober,2,0,0,23%,14,2,271.08,20,20,272,331,621
1,4/10,6:00 PM,45,sober,1,2,0,30%,19,1,396.73,123,30,247,444,998
2,4/10,6:00 PM,38,high,3,0,0,30%,32,1,607.80,71,60,176,322,1109
3,4/10,7:00 PM,30,high,1,3,0,18%,19,1,714.16,244,10,238,330,4726
4,4/10,7:00 PM,16,high,3,1,1,58%,42,18,1140.00,584,150,365,668,2070


,Date,Time of Day,Placed,Mental State,Eliminations,Assists,Revives,Accuracy,Hits,Head Shots,Distance Traveled,Materials Gathered,Materials Used,Damage Taken,Damage to Players,Damage to Structures
82,17-Apr,12:00 AM,2,high,8,2,0,31%,97,8,3770.00,1750,400,576,1492,4593
83,17-Apr,12:00 AM,24,high,4,0,0,20%,12,4,222.54,94,0,96,478,2262
84,17-Apr,1:00 PM,30,sober,1,2,0,28%,40,4,267.00,78,30,148,375,1196
85,17-Apr,2:00 PM,18,sober,3,2,1,26%,25,3,954.43,370,60,354,475,1168
86,17-Apr,2:00 PM,39,sober,1,0,0,31%,12,1,226.67,75,0,120,338,1105


(87, 16)

,Placed,Eliminations,Assists,Revives,Hits,Head Shots,Distance Traveled,Materials Gathered,Materials Used,Damage Taken,Damage to Players,Damage to Structures
count,87.000000,87.000000,87.000000,87.000000,87.000000,87.000000,87.000000,87.000000,87.000000,87.000000,87.000000,87.00000
mean,22.045977,2.517241,1.482759,0.402299,29.735632,4.747126,1137.146322,386.574713,122.712644,244.172414,581.770115,3170.91954
std,13.145791,1.885453,1.388173,0.738631,22.093596,5.777298,1110.843642,569.978062,225.170370,124.937399,354.172396,3458.54107
min,1.000000,0.000000,0.000000,0.000000,1.000000,0.000000,17.860000,0.000000,0.000000,19.000000,60.000000,175.00000
25%,15.000000,1.000000,0.000000,0.000000,12.000000,1.000000,398.280000,75.000000,20.000000,154.000000,328.500000,1094.00000
50%,21.000000,2.000000,1.000000,0.000000,27.000000,3.000000,638.170000,164.000000,50.000000,209.000000,481.000000,2029.00000
75%,28.500000,3.000000,2.000000,1.000000,38.000000,6.500000,1575.000000,418.500000,145.000000,316.500000,750.000000,3707.00000
max,66.000000,8.000000,6.000000,4.000000,105.000000,33.000000,4460.000000,3002.000000,1740.000000,677.000000,1507.000000,18026.00000


Identifica cuántas columnas son numéricas y cuántas son categóricas

In [124]:
col_numericas = fortnite.select_dtypes(include='number').columns
col_categoricas = fortnite.select_dtypes(include='object').columns

print(f'Columnas numéricas: {len(col_numericas)}')
print(f'Columnas categóricas: {len(col_categoricas)}')

Columnas numéricas: 12
Columnas categóricas: 4


C:\Users\User\AppData\Local\Temp\ipykernel_14752\2172552209.py:2: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  col_categoricas = fortnite.select_dtypes(include='object').columns


Muestra el nombre y conteo de valores únicos de las columnas categóricas más importantes

In [125]:
for col in col_categoricas:
    print(f'\nColumna: {col}')
    print(f'Valores únicos: {fortnite[col].nunique()}')
    print(fortnite[col].value_counts())


Columna: Date
Valores únicos: 7
Date
4/14      24
4/13      17
4/10      14
4/15      13
4/11      12
17-Apr     5
16-Apr     2
Name: count, dtype: int64

Columna: Time of Day
Valores únicos: 13
Time of Day
2:00 PM     18
1:00 PM     13
8:00 PM     11
7:00 PM     10
6:00 PM      7
3:00 PM      6
4:00 PM      5
9:00 PM      4
11:00 PM     4
5:00 PM      3
12:00 PM     2
10:00 PM     2
12:00 AM     2
Name: count, dtype: int64

Columna: Mental State
Valores únicos: 2
Mental State
sober    45
high     42
Name: count, dtype: int64

Columna: Accuracy
Valores únicos: 38
Accuracy
25%    6
17%    6
23%    4
27%    4
20%    4
26%    4
16%    4
18%    3
10%    3
21%    3
22%    3
24%    3
28%    3
34%    3
38%    3
30%    2
39%    2
35%    2
13%    2
4%     2
36%    2
11%    2
31%    2
58%    1
46%    1
6%     1
12%    1
8%     1
29%    1
90%    1
44%    1
43%    1
7%     1
53%    1
45%    1
42%    1
64%    1
33%    1
Name: count, dtype: int64


### 2.2 — Filtrado y selección

Selecciona únicamente las columnas numéricas del dataset y guárdalas en un nuevo DataFrame llamado df_fortnite_num

In [126]:
df_fortnite_num = fortnite.select_dtypes(include='number')

display(df_fortnite_num)

,Placed,Eliminations,Assists,Revives,Hits,Head Shots,Distance Traveled,Materials Gathered,Materials Used,Damage Taken,Damage to Players,Damage to Structures
0,27,2,0,0,14,2,271.08,20,20,272,331,621
1,45,1,2,0,19,1,396.73,123,30,247,444,998
2,38,3,0,0,32,1,607.80,71,60,176,322,1109
3,30,1,3,0,19,1,714.16,244,10,238,330,4726
4,16,3,1,1,42,18,1140.00,584,150,365,668,2070
...,...,...,...,...,...,...,...,...,...,...,...,...
82,2,8,2,0,97,8,3770.00,1750,400,576,1492,4593
83,24,4,0,0,12,4,222.54,94,0,96,478,2262
84,30,1,2,0,40,4,267.00,78,30,148,375,1196
85,18,3,2,1,25,3,954.43,370,60,354,475,1168


Filtra y muestra solo las filas donde el jugador haya obtenido más de 5 kills

In [127]:
mas_5_kills = df_fortnite_num[df_fortnite_num['Eliminations'] > 5]

display(mas_5_kills)

,Placed,Eliminations,Assists,Revives,Hits,Head Shots,Distance Traveled,Materials Gathered,Materials Used,Damage Taken,Damage to Players,Damage to Structures
20,1,7,0,0,63,9,3320.0,541,530,180,1399,17151
28,8,6,2,0,59,10,2560.0,732,320,317,1444,7028
30,2,8,0,1,34,7,3790.0,3002,1740,321,1166,10594
37,13,6,0,0,92,22,1500.0,894,160,257,1253,4871
38,1,6,3,1,46,7,3830.0,1907,530,193,1308,18026
41,2,6,2,0,54,5,4050.0,2553,680,400,1383,4945
68,13,6,2,0,38,7,1890.0,1560,140,367,1005,9540
69,5,6,3,0,29,3,3350.0,421,230,366,764,11077
82,2,8,2,0,97,8,3770.0,1750,400,576,1492,4593


Filtra las partidas donde el jugador quedó en el top 10 (posición ≤ 10)

In [128]:
top_10 = df_fortnite_num[df_fortnite_num['Placed'] <= 10]

display(top_10)

,Placed,Eliminations,Assists,Revives,Hits,Head Shots,Distance Traveled,Materials Gathered,Materials Used,Damage Taken,Damage to Players,Damage to Structures
15,4,5,4,4,35,3,3230.0,715,420,200,894,9494
16,5,3,6,1,32,11,2780.0,384,370,264,787,3589
20,1,7,0,0,63,9,3320.0,541,530,180,1399,17151
28,8,6,2,0,59,10,2560.0,732,320,317,1444,7028
30,2,8,0,1,34,7,3790.0,3002,1740,321,1166,10594
36,10,3,3,0,48,5,1810.0,512,150,567,651,2371
38,1,6,3,1,46,7,3830.0,1907,530,193,1308,18026
41,2,6,2,0,54,5,4050.0,2553,680,400,1383,4945
42,6,1,1,0,8,0,2810.0,288,280,352,256,5036
52,5,4,5,1,105,33,2330.0,687,50,671,1507,5063


Usa .loc para mostrar la fila con el mayor número de kills en toda la historia del dataset

In [129]:
max_kills = fortnite['Eliminations'].idxmax()

fortnite.loc[max_kills]

Date                       4/13
Time of Day             2:00 PM
Placed                        2
Mental State              sober
Eliminations                  8
Assists                       0
Revives                       1
Accuracy                    17%
Hits                         34
Head Shots                    7
Distance Traveled        3790.0
Materials Gathered         3002
Materials Used             1740
Damage Taken                321
Damage to Players          1166
Damage to Structures      10594
Name: 30, dtype: object

### 2.3 — Agrupación y agregación

Agrupa los datos por el tipo de modo de juego (si existe la columna) o por alguna columna categórica relevante

In [130]:
grupos = fortnite.groupby('Mental State')

Calcula el promedio de kills, daño infligido y tiempo de partida por cada grupo

In [131]:
promedios = round(grupos[['Eliminations', 'Damage to Players', 'Placed']].mean(), 2)

display(promedios)

,Eliminations,Damage to Players,Placed
Mental State,,,
high,2.50,557.55,22.17
sober,2.53,604.38,21.93


Muestra el grupo con el mejor rendimiento promedio

In [132]:
mejor_grupo = promedios['Eliminations'].idxmax()

print('Grupo con mejor rendimiento promedio:', mejor_grupo)

Grupo con mejor rendimiento promedio: sober


### 2.4 — Análisis y preguntas

**P1. ¿Cuál es el número máximo, mínimo y promedio de kills por partida?**

In [133]:
maximo = fortnite['Eliminations'].max()
minimo = fortnite['Eliminations'].min()
promedio = fortnite['Eliminations'].mean()

print(f'Máximo: {maximo}')
print(f'Mínimo: {minimo}')
print(f'Promedio: {promedio:.2f}')

Máximo: 8
Mínimo: 0
Promedio: 2.52


*El resultado indica que no es tan sencillo llegar a tener un número alto de kills, por lo que se debe tener práctica para lograrlo*

**P2. ¿Existe correlación entre el número de kills y el tiempo que el jugador sobrevive? Usa .corr() para responder**

![correlacion](correlacion.png)

In [134]:
correlacion = fortnite['Eliminations'].corr(fortnite['Placed'])

print(f'Correlación: {correlacion:.5f}')

Correlación: -0.62070


*Este resultado tiene sentido porque mientras más kills menor es el 'Placed' (tiempo de juego), no tiene correlación negativa perfecta (-1) porque no siempre se debe tener alguna/s kill/s para ganar una partida*

**P3. ¿En qué porcentaje de partidas el jugador terminó en el top 1 (ganó la partida)?**

In [135]:
porcentaje_ganador = (fortnite['Placed'] == 1).mean() * 100

print(f'Porcentaje: {porcentaje_ganador:.2f}%')

Porcentaje: 3.45%


*Este resultado indica que de las 87 partidas solo se ganaron el 3.45% (3), demostrando lo complicado que es quedar en primer lugar*